cleanlab/multilabel_classification/dataset.py

```diff
@@ -0,0 +1,369 @@
+    dataset. So a score of 1 suggests your data is perfectly labeled and a score of 0.5 suggests
+    half of the examples in the dataset may be incorrectly labeled. Thus, a higher
+    score implies a higher quality dataset.
+
+    This method works by providing any one (and only one) of the following inputs:
+
+    1. ``labels`` and ``pred_probs``, or
+    2. ``joint`` and ``num_examples``, or
+    3. ``confident_joint``
+
+    Only provide **exactly one of the above input options**, do not provide a combination.
+
+    **Parameters**: For parameter info, see the docstring of :py:func:`find_overlapping_classes <cleanlab.dataset.find_overlapping_classes>`.
+
+    Returns
+    -------
+    health_score : float
+        A score between 0 and 1, where 1 implies all labels in the dataset are estimated to be correct.
+        A score of 0.5 implies that half of the dataset's labels are estimated to have issues.
+    """
+    num_classes = get_num_classes(labels=labels, pred_probs=pred_probs, multi_label=True)
+    class_names = list(range(num_classes))
+    num_examples = _get_num_examples_multilabel(labels=labels, confident_joint=confident_joint)
+    issues_df = common_multilabel_issues(
+        labels=labels, pred_probs=pred_probs, class_names=class_names, confident_joint=joint
+    )
+    return sum(issues_df["Num Examples"]) / num_examples
+
+
+def health_summary(
+    labels=None,
+    pred_probs=None,
+    *,
+    class_names=None,
+    num_examples=None,
+    joint=None,
+    confident_joint=None,
+    verbose=True,
+) -> dict:
+    """Prints a health summary of your datasets including useful statistics like:
+
+    * The classes with the most and least label issues
+    * Overall data label quality health score statistics for your dataset
+
+    This method works by providing any one (and only one) of the following inputs:
+
+    1. ``labels`` and ``pred_probs``, or
+    2. ``joint`` and ``num_examples``, or
+    3. ``confident_joint``
+
+    Only provide **exactly one of the above input options**, do not provide a combination.
+
+    **Parameters**: For parameter info, see the docstring of :py:func:`find_overlapping_classes <cleanlab.dataset.find_overlapping_classes>`.
+
+    Returns
+    -------
+    summary : dict
+        A dictionary containing keys (see the corresponding functions' documentation to understand the values):
+
+        - ``"overall_label_health_score"``, corresponding to :py:func:`overall_label_health_score <cleanlab.dataset.overall_label_health_score>`
+        - ``"joint"``, corresponding to :py:func:`estimate_joint <cleanlab.count.estimate_joint>`
+        - ``"classes_by_label_quality"``, corresponding to :py:func:`rank_classes_by_label_quality <cleanlab.dataset.rank_classes_by_label_quality>`
+        - ``"overlapping_classes"``, corresponding to :py:func:`find_overlapping_classes <cleanlab.dataset.find_overlapping_classes>`
+    """
```

Comment:
a decent amount of the code for this method is copy/pasted from the version of this method in cleanlab.dataset.

Consider moving the common parts of the code to a helper method in utils/, so you that you can reduce the amount of copy/paste. But let's not worry about doing this if it's not easy, only do this if it is straightforward!